rocksdb_replicator/replicated_db.cpp

```diff
@@ -85,49 +85,61 @@ rocksdb::Status RocksDBReplicator::ReplicatedDB::Write(
     throw ReturnCode::WRITE_TO_SLAVE;
   }
 
+  auto write_begin = GetCurrentTimeMs();
+
   incCounter(kReplicatorWriteBytes, updates->GetDataSize(), db_name_);
 
   auto ms = GetCurrentTimeMs();
   updates->PutLogData(rocksdb::Slice(reinterpret_cast<const char*>(&ms),
                                      sizeof(ms)));
-  auto start = GetCurrentTimeMs();
+  auto write_leader_begin = GetCurrentTimeMs();
   auto status = db_wrapper_->WriteToLeader(options, updates);
-  auto end = GetCurrentTimeMs();
-  logMetric(kReplicatorWriteMs, start < end ? end - start : 0, db_name_);
-  
+  auto write_leader_end = GetCurrentTimeMs();
+  auto write_leader_time = write_leader_begin < write_leader_end ? write_leader_end - write_leader_begin : 0;
+  logMetric(kReplicatorWriteToLeaderMs, write_leader_time, db_name_);
+
+  if (!status.ok()) {
+    incCounter(kReplicatorWriteLeaderFailure, 1, db_name_);
```

Comment:
I'm fine if you want to keep this, but you can always use the `count` of a metric to get the counter. You don't need a counter _and_ a metric.

Reply:
yeah, it took me a while to realize this when looking at our rockstore metrics. I still prefer separate counters to be explicit, and consistent with other success/failure counters. Otherwise the naming of the metrics could confuse people. Plus there is little perf concern here given this is in the failure path.